src/aria2p/utils.py

```diff
@@ -127,3 +130,111 @@ def get_version():
         return Version("0.0.0")
     else:
         return Version(distribution.version)
+
+
+class CustomConfig:
+    config = configparser.ConfigParser()
+
+    # TODO Parse default configuration (fallback values) from file.
+    # TODO Create default configuration file during build.
+    #       eg. $HOME/.config/aria2p/config.ini
+
+    # Fallback values
+    DEFAULT_VALUES = {
+        "AUTOCLEAR": "c",
+        "CANCEL": "escape",
+        "ENTER": "enter",
+        "FILTER": "F4",
+        "FOLLOW_ROW": "F",
+        "HELP": "F1",
+        "MOVE_DOWN": "down",
+        "MOVE_DOWN_STEP": "J",
+        "MOVE_END": "end",
+        "MOVE_HOME": "home",
+        "MOVE_LEFT": "left",
+        "MOVE_RIGHT": "right",
+        "MOVE_UP": "up",
+        "MOVE_UP_STEP": "K",
+        "NEXT_SORT": ">",
+        "PREVIOUS_SORT": "<",
+        "PRIORITY_DOWN": "F8",
+        "PRIORITY_UP": "F7",
+        "QUIT": "F10",
+        "REMOVE_ASK": "delete",
+        "REVERSE_SORT": "I",
+        "SEARCH": "F3",
+        "RETRY": "r",
+        "RETRY_ALL": "R",
+        "SELECT_SORT": "F6",
+        "SETUP": "F2",
+        "TOGGLE_EXPAND_COLLAPSE_ALL": "X",
+        "TOGGLE_EXPAND_COLLAPSE": "x",
+        "TOGGLE_RESUME_PAUSE": "space",
+        "TOGGLE_RESUME_PAUSE_ALL": "P",
+        "TOGGLE_SELECT": "s",
+        "UN_SELECT_ALL": "U",
+        # [colors]
+        "BRIGHT_HELP": "CYAN BOLD BLACK",
+        "FOCUSED_HEADER": "BLACK NORMAL CYAN",
+        "FOCUSED_ROW": "BLACK NORMAL CYAN",
+        "HEADER": "BLACK NORMAL GREEN",
+        "METADATA": "WHITE UNDERLINE BLACK",
+        "SIDE_COLUMN_FOCUSED_ROW": "BLACK NORMAL CYAN",
+        "SIDE_COLUMN_HEADER": "BLACK NORMAL GREEN",
+        "SIDE_COLUMN_ROW": "WHITE NORMAL BLACK",
+        "STATUS_ACTIVE": "CYAN NORMAL BLACK",
+        "STATUS_COMPLETE": "GREEN NORMAL BLACK",
+        "STATUS_ERROR": "RED BOLD BLACK",
+        "STATUS_PAUSED": "YELLOW NORMAL BLACK",
+        "STATUS_WAITING": "WHITE BOLD BLACK",
+    }
+
+    """
+    Get values of colors and modes for ascimatics.screen module
+    """
+    COLOUR_MAPPINGS = {
+        "BLACK": Screen.COLOUR_BLACK,
+        "WHITE": Screen.COLOUR_WHITE,
+        "RED": Screen.COLOUR_RED,
+        "CYAN": Screen.COLOUR_CYAN,
+        "YELLOW": Screen.COLOUR_YELLOW,
+        "BLUE": Screen.COLOUR_BLUE,
+        "GREEN": Screen.COLOUR_GREEN,
+    }
+
+    MODE_MAPPINGS = {
+        "NORMAL": Screen.A_NORMAL,
+        "BOLD": Screen.A_BOLD,
+        "UNDERLINE": Screen.A_UNDERLINE,
+        "REVERSE": Screen.A_REVERSE,
+    }
+
+    def __init__(self):
+        """
+        Parse configuration file if it exist in the default config path
+        If it doesn't exist default values are parsed
+        """
+        self.config["DEFAULT"] = self.DEFAULT_VALUES
+        self.config["keys"] = {}
+        self.config["colors"] = {}
+
+        config_path = Path.home() / ".config" / "aria2p" / "config.ini"
```

Comment:
We should try to respect the XDG spec and use the value of the `XDG_CONFIG_HOME` environment variable if defined.